### PR TITLE
Fix Histogram

### PR DESCRIPTION
**Pull request by pnor**
Change the Histogram in the GymDetailView to display more accurate labels as described by the graph. Also fix an indexing bug where tapping the histogram would show a label describing the bar to its left.
Screen Shot 2019-09-16 at 11 25 20 PM

Screen Shot 2019-09-16 at 11 25 35 PM

Screen Shot 2019-09-16 at 11 25 46 PM

Screen Shot 2019-09-16 at 11 25 57 PM

Screen Shot 2019-09-16 at 11 26 44 PM

### CODE REVIEW

File: Uplift/Views/Histogram.swift [hunk 1]
```diff
@@ -14,64 +14,64 @@ import UIKit
...
-    var hours: DailyGymHours!
-    var openHour: Int!
-
-    var bottomAxis: UIView!
-    var bottomAxisTicks: [UIView]!
-
+    private var selectedIndex: Int!
+    private var selectedLine: UIView!
+    private var selectedTime: UILabel!
+
```
Comment (kevinchan159) [inline Uplift/Views/Histogram.swift]: delete whitespace

File: Uplift/Views/Histogram.swift [hunk 2]
```diff
@@ -14,64 +14,64 @@ import UIKit
...
+    private var selectedIndex: Int!
+    private var selectedLine: UIView!
+    private var selectedTime: UILabel!
+    
+    private let highThreshold = 57
+    private let mediumThreshold = 25
+    private let secondsPerHour: Double = 3600.0
+    private let timeDescriptors = ["Not too busy", "A little busy", "As busy as it gets"]
+    private var selectedTimeDescriptor: UILabel!
+
```
Comment (kevinchan159) [inline Uplift/Views/Histogram.swift]: delete whitespace

File: Uplift/Views/Histogram.swift [hunk 3]
```diff
@@ -14,64 +14,64 @@ import UIKit
...
+    
+    private let highThreshold = 57
+    private let mediumThreshold = 25
+    private let secondsPerHour: Double = 3600.0
+    private let timeDescriptors = ["Not too busy", "A little busy", "As busy as it gets"]
+    private var selectedTimeDescriptor: UILabel!
+    
+    private var hours: DailyGymHours!
+    private var openHour: Int!
+
```
Comment (kevinchan159) [inline Uplift/Views/Histogram.swift]: same

File: Uplift/Views/Histogram.swift [hunk 4]
```diff
@@ -14,64 +14,64 @@ import UIKit
...
+    private let secondsPerHour: Double = 3600.0
+    private let timeDescriptors = ["Not too busy", "A little busy", "As busy as it gets"]
+    private var selectedTimeDescriptor: UILabel!
+    
+    private var hours: DailyGymHours!
+    private var openHour: Int!
+    
+    private var bottomAxis: UIView!
+    private var bottomAxisTicks: [UIView]!
+
```
Comment (kevinchan159) [inline Uplift/Views/Histogram.swift]: same

File: Uplift/Views/Histogram.swift [hunk 5]
```diff
@@ -14,64 +14,64 @@ import UIKit
...
+    private var bottomAxis: UIView!
+    private var bottomAxisTicks: [UIView]!
+    
     init(frame: CGRect, data: [Int], todaysHours: DailyGymHours) {
+        openHour = Calendar.current.component(.hour, from: todaysHours.openTime)
         super.init(frame: frame)
         self.data = data
         self.hours = todaysHours
-
+
```
Comment (kevinchan159) [inline Uplift/Views/Histogram.swift]: same

File: Uplift/Views/Histogram.swift [hunk 6]
```diff
@@ -14,64 +14,64 @@ import UIKit
...
         self.data = data
         self.hours = todaysHours
-
+        
         // AXIS
         bottomAxis = UIView()
         bottomAxis.backgroundColor = .fitnessLightGrey
         addSubview(bottomAxis)
-
+
```
Comment (kevinchan159) [inline Uplift/Views/Histogram.swift]: same

**Comment by ywsang** [comment]
still need to review, but saw cameron’s comment — i fixed that bug in the PR i just merged so if you merge in master that should be resolved

File: Uplift/Views/Histogram.swift [hunk 7]
```diff
@@ -214,6 +215,19 @@ class Histogram: UIView {
...
+        if data[selectedIndex + openHour] < mediumThreshold {
+            selectedTimeDescriptor.text = timeDescriptors[0]
+        } else if data[selectedIndex + openHour] < highThreshold {
+            selectedTimeDescriptor.text = timeDescriptors[1]
+        } else {
+            selectedTimeDescriptor.text = timeDescriptors[2]
+        }
+        selectedTimeDescriptor.sizeToFit()
+        
+        selectedTime.text = hours.openTime.addingTimeInterval( Double(selectedIndex) * secondsPerHour ).getStringOfDatetime(format: "ha :")
```
Comment (ywsang) [inline Uplift/Views/Histogram.swift]: make sure you resolve this conflict and remove the <<<<< HEAD and >>>>>> master flags!

File: .vscode/settings.json [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (ywsang) [inline .vscode/settings.json]: should this file be here?

File: Uplift/Views/Histogram.swift [hunk 9]
```diff
@@ -214,15 +215,17 @@ class Histogram: UIView {
...
-                if data[selectedIndex + openHour - 1] < mediumThreshold {
-                    selectedTimeDescriptor.text = timeDescriptors[0]
-                } else if data[selectedIndex + openHour - 1] < highThreshold {
-                    selectedTimeDescriptor.text = timeDescriptors[1]
-                } else {
-                    selectedTimeDescriptor.text = timeDescriptors[2]
-                }
-                selectedTimeDescriptor.sizeToFit()
+        // update selectedTime and the descriptor
+        if data[selectedIndex + openHour] < mediumThreshold {
```
Comment (chalo2000) [inline Uplift/Views/Histogram.swift]: duplicate code!!! (103-108)
i would make a function

Comment (pnor) [reply Uplift/Views/Histogram.swift]: fixed

**Pull request merged**